### DISCUSSION TITLE
AGW: Separating Build from Install ?

### DISCUSSION THREAD

**Discussion by ferrieux** [magma/magma#15319] [Q&A] [answered]
Hi,

The two documented methods for AGW "build" (Bare metal and Docker) don't seem to allow separating the Build from the Deploy steps.
Worse, both variants thoroughly reconfigure the host system before doing anything.
Trying to go into (say) lte/gateway and running "make" also fails catastrophically, for lack of a bunch of environment variables it seems.

What is the idiomatic method to *just compile* without wrecking the host system ?

**Comment by panyogesh** [comment]
@ferrieux : There are 2 main points for which changes are are required: 
1. Changing the name of interface to be inline with MME/AMF (SCTP) and upf.  This is optional in case if we know which configuration file to tune.
2. DNS related changes.  This can be restored in case if the connectivity is there.

Is there any other specific host parameter which is causing trouble. Please let me know. We tested the Point-1 (retaining the original interface name) and it seems to be working fine.

Let me know if the preferred way is dockers. I can share the steps for the same.

**Reply by ferrieux** [reply]
Sorry, you're diving into details I'm not familiar with. What I mean is: if Magma aims to be "carrier grade" some day, one obvious prerequisite is to be able to compile on one machine and deploy on N other ones.  An equally important must-have is _simple_ instructions like "./configure;make;make deb" (on the compile machine) and "apt install ./xxx.deb" (on the deploy machines), instead of several redundant-or-inconsistent, out-of-date web pages with dozens of tricky and non-deterministic manual steps.

Note this applies to all components of Magma, as the install instructions are also a mess for the orchestrator and fegs: being carrier-grade means that someone who (1) hasn't spent months following the technical evolution of Magma internals and (2) doesn't  want to spend anxious hours in an error-prone manual procedure (with very cryptic feedback from the logs if something went wrong)
, must be able to deploy.

Before that, Magma is very interesting - yet unfinished - lab experiment.

**Comment by lucasgonze** [comment]
This is a documentation problem. We should write up instructions on how to build without installing.

Yogesh will check documentation and possibly create a ticket.

**Comment by panyogesh** [comment, marked as answer]
I have created a new ticket for the same [#15364]. Kindly review and add the comments as applicable. 
Will try to internally brainstorm to come up with timelines.
